diff --git a/lisp/cus-start.el b/lisp/cus-start.el
index fa12045db8..fe0319e2b4 100644
--- a/lisp/cus-start.el
+++ b/lisp/cus-start.el
@@ -601,6 +601,7 @@ Leaving \"Default\" unchecked is equivalent with specifying a default of
 	     (mac-pass-command-to-system mac boolean "22.1")
 	     (mac-pass-control-to-system mac boolean "22.1")
 	     (mac-drawing-use-gcd mac boolean "24.4")
+             (mac-use-title-bar mac boolean "25.2")
 	     (mac-frame-tabbing mac (choice (const automatic) (const inverted)
                                             (const nil) (const t)) "24.5")
 	     (mac-ignore-accessibility mac boolean "25.3")
diff --git a/src/macappkit.m b/src/macappkit.m
index 28c6b38f00..0ae299dcf3 100644
--- a/src/macappkit.m
+++ b/src/macappkit.m
@@ -2495,9 +2495,13 @@ - (void)setupWindow
       if (!self.shouldBeTitled)
 	windowStyle = NSWindowStyleMaskBorderless;
       else
-	windowStyle = (NSWindowStyleMaskTitled | NSWindowStyleMaskClosable
-		       | NSWindowStyleMaskMiniaturizable
-		       | NSWindowStyleMaskResizable);
+        if (mac_use_title_bar)
+          windowStyle = (NSWindowStyleMaskTitled | NSWindowStyleMaskClosable
+                         | NSWindowStyleMaskMiniaturizable
+                         | NSWindowStyleMaskResizable);
+        else
+          windowStyle = (NSWindowStyleMaskBorderless | NSWindowStyleMaskResizable
+                         | NSWindowStyleMaskMiniaturizable);
     }
   else
     windowStyle = NSWindowStyleMaskBorderless;
diff --git a/src/macterm.c b/src/macterm.c
index 936a57d2ff..13d278eebc 100644
--- a/src/macterm.c
+++ b/src/macterm.c
@@ -6225,6 +6225,16 @@ tabbing is disallowed and preferred, respectively.
 This variable has no effect on OS X 10.11 and earlier.  */);
   Vmac_frame_tabbing = Qautomatic;
 
+  DEFVAR_BOOL ("mac-use-title-bar", mac_use_title_bar,
+     doc: /*Non-nil means to include the title bar on frames.  Nil means to
+omit the title bar on OSX >= 10.7.  This variable is ignored on OSX <
+10.7. Note that the first frame per emacs session (unless it is an
+emacsclient) will use the default value of this variable (nil--do not
+include a title bar), ignoring (!) the user-set value of this
+variable. This is almost certainly due to a bug in the patch adding
+this variable. Default is nil.*/);
+  mac_use_title_bar = false;
+
   DEFVAR_BOOL ("mac-ignore-accessibility", mac_ignore_accessibility,
     doc: /* Non-nil means to ignore the system accessibility feature.
 Emacspeak users may want to set this variable so it may not interfere
